numba/cpython/setobj.py

```diff
@@ -1089,6 +1098,9 @@ def _copy_payload(self, src_payload):
                     cgutils.raw_memcpy(builder, payload.entries,
                                        src_payload.entries, nentries,
                                        entry_size)
+                    with src_payload._iterate() as loop:
+                        entry = loop.entry
+                        context.nrt.incref(builder, self._ty.dtype, entry.key)
```

Comment:
Is this case tested? I think this is correct, but I would add a comment about whey you aren't using `_add_key`.